### PR TITLE
[host network] keep backward compatibility in release-1.0

### PR DESCRIPTION
**Pull request by cofyc**
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Helm charts change
 - Has Go code change
 - Has CI related scripts change
 - Has documents change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
bugfix: don't set DNSPolicy if hostNetwork is not enabled to keep backward compatibility
 ```

### CODE REVIEW

File: pkg/manager/member/pd_member_manager.go [hunk 1]
```diff
@@ -522,6 +553,7 @@ func (pmm *pdMemberManager) getNewPDSetForTidbCluster(tc *v1alpha1.TidbCluster)
 				},
 			},
 		})
+		podSpec.Containers[0].Env = env
```
Comment (weekface) [inline pkg/manager/member/pd_member_manager.go]: only add `env` when using `hostNetwork`?

Comment (cofyc) [reply pkg/manager/member/pd_member_manager.go]: ah.. sorry, fixed

**Pull request merged**